test/document_stores/test_faiss.py

```diff
@@ -8,294 +9,266 @@
-
-    # test saving the index
-    document_store.save(index_path=tmp_path / "haystack_test_faiss", config_path=tmp_path / "custom_path.json")
-
-    # clear existing faiss_index
-    document_store.faiss_indexes[document_store.index].reset()
-
-    # test faiss index is cleared
-    assert document_store.faiss_indexes[document_store.index].ntotal == 0
-
-    # test loading the index
-    new_document_store = FAISSDocumentStore.load(
-        index_path=tmp_path / "haystack_test_faiss", config_path=tmp_path / "custom_path.json"
-    )
-
-    # check faiss index is restored
-    assert new_document_store.faiss_indexes[document_store.index].ntotal == len(DOCUMENTS)
-    # check if documents are restored
-    assert len(new_document_store.get_all_documents()) == len(DOCUMENTS)
-    # Check if the init parameters are kept
-    assert not new_document_store.progress_bar
-
-    # test saving and loading the loaded faiss index
-    new_document_store.save(tmp_path / "haystack_test_faiss", config_path=tmp_path / "custom_path.json")
-    reloaded_document_store = FAISSDocumentStore.load(
-        tmp_path / "haystack_test_faiss", config_path=tmp_path / "custom_path.json"
-    )
-
-    # check faiss index is restored
-    assert reloaded_document_store.faiss_indexes[document_store.index].ntotal == len(DOCUMENTS)
-    # check if documents are restored
-    assert len(reloaded_document_store.get_all_documents()) == len(DOCUMENTS)
-    # Check if the init parameters are kept
-    assert not reloaded_document_store.progress_bar
-
-    # test loading the index via init
-    new_document_store = FAISSDocumentStore(
-        faiss_index_path=tmp_path / "haystack_test_faiss", faiss_config_path=tmp_path / "custom_path.json"
-    )
-
-    # check faiss index is restored
-    assert new_document_store.faiss_indexes[document_store.index].ntotal == len(DOCUMENTS)
-    # check if documents are restored
-    assert len(new_document_store.get_all_documents()) == len(DOCUMENTS)
-    # Check if the init parameters are kept
-    assert not new_document_store.progress_bar
-
-
-@pytest.mark.skipif(sys.platform in ["win32", "cygwin"], reason="Test with tmp_path not working on windows runner")
-def test_faiss_index_mutual_exclusive_args(tmp_path):
-    with pytest.raises(ValueError):
-        FAISSDocumentStore(
-            sql_url=f"sqlite:////{tmp_path/'haystack_test.db'}",
-            faiss_index_path=f"{tmp_path/'haystack_test'}",
-            isolation_level="AUTOCOMMIT",
-        )
+class TestFAISSDocumentStore(DocumentStoreBaseTestAbstract):
+    index_name = __name__
 
-    with pytest.raises(ValueError):
-        FAISSDocumentStore(
-            f"sqlite:////{tmp_path/'haystack_test.db'}",
-            faiss_index_path=f"{tmp_path/'haystack_test'}",
+    @pytest.fixture
+    def ds(self, tmp_path):
+        return FAISSDocumentStore(
+            sql_url=f"sqlite:///{tmp_path}/haystack_test.db",
+            return_embedding=True,
+            index=self.index_name,
```

Comment:
Why not the default?

Reply:
it comes from Elastic/Opensearch, where we prefer to use a dedicated index for tests to avoid clashing with local instances of the database servers

Reply:
Right, but FAISS instances should be fully isolated regardless. The FAISS index is in memory and the SQLite database is in a temp path. So there's no risk for collisions. But that's a nitpick anyway

Reply:
Right, we have the file name here... I can remove it (probably have the same in the Sql tests, we can uniform it later)